### PR TITLE
Fixed titles mode

### PR DESCRIPTION
**Pull request by oroom**
Title shift mode added (TitleShiftMode). It can be used to specify other titles animation behavior then centered position.
.paged mode added. It works almost same with .center but adjusts title position to match title border with titles superview border.

Used only for X axis titles layout.

### CODE REVIEW

File: Source/TitleScrollView.swift [hunk 1]
```diff
@@ -13,6 +13,8 @@ public protocol TitleConfigurable: class {
     var items: [TitleItem] { get }
     var alignment: TitleViewAlignment { get set }
     var position: TitleViewPosition { get set }
+    /// Defigns how titles shifted while switching main content.
```
Comment (ededovets) [inline Source/TitleScrollView.swift]: Please put comment to describe the behavior we can then add to Readme

**Comment by oroom** [comment]
@ededovets Code updated to fix bugs. Comments updated to clarify TitleShiftMode options. Example project updated to demonstrate behavior in horizontal example. But current example almost don't show real difference 😕.

**Comment by ededovets** [comment]
@oroom is it possible to make fast change for current TitleView style to demonstrate the difference?

**Comment by ededovets** [comment]
@oroom Also doest this mode work for Vertical mode? If not then why you add switcher to Vertical sample page?

**Comment by pkondrashkov** [comment]
@oroom
I have checked this mode on large text titles and it is not clear what was the purpose of this mode.
When you add titles with long text it shows the previous title on the screen but it works only with left to right scroll direction, if we scroll from right to left it looks like title view is broken because we don't see the previous title item, only the next.  
So I assume it work only in one direction which looks like not finished \ broken.

**Pull request closed without merging**